File: backend/src/models/Restaurant.js
const mongoose = require('mongoose');

const RestaurantSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    description: { type: String, default: '' },
    cuisine: { type: String, default: '' },
    imageUrl: { type: String, default: '' },
    city: { type: String, default: '' },
    rating: { type: Number, default: 4.5 },
    priceLevel: { type: String, default: '₸₸' },
    deliveryTime: { type: String, default: '25-35 мин' },
    minOrder: { type: Number, default: 1500 },
    deliveryFee: { type: Number, default: 0 },
    tags: { type: [String], default: [] },
    isActive: { type: Boolean, default: true }
  },
  { timestamps: true }
);

module.exports = mongoose.model('Restaurant', RestaurantSchema);

File: frontend/lib/models/restaurant.dart
class Restaurant {
  Restaurant({
    required this.id,
    required this.name,
    required this.description,
    required this.cuisine,
    required this.imageUrl,
    required this.city,
    required this.rating,
    required this.priceLevel,
    required this.deliveryTime,
    required this.minOrder,
    required this.deliveryFee,
    required this.tags,
  });

  final String id;
  final String name;
  final String description;
  final String cuisine;
  final String imageUrl;
  final String city;
  final double rating;
  final String priceLevel;
  final String deliveryTime;
  final int minOrder;
  final int deliveryFee;
  final List<String> tags;

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['_id'] as String,
      name: json['name'] as String,
      description: (json['description'] ?? '') as String,
      cuisine: (json['cuisine'] ?? '') as String,
      imageUrl: (json['imageUrl'] ?? '') as String,
      city: (json['city'] ?? '') as String,
      rating: (json['rating'] ?? 0).toDouble(),
      priceLevel: (json['priceLevel'] ?? '₸₸') as String,
      deliveryTime: (json['deliveryTime'] ?? '') as String,
      minOrder: (json['minOrder'] ?? 0) as int,
      deliveryFee: (json['deliveryFee'] ?? 0) as int,
      tags: (json['tags'] as List? ?? []).map((tag) => tag.toString()).toList(),
    );
  }
}
